src/engine/sparqlExpressions/StringExpressions.cpp

```diff
@@ -399,6 +399,31 @@ class ConcatExpression : public detail::VariadicExpression {
 using EncodeForUriExpression =
     StringExpressionImpl<1, decltype(encodeForUriImpl)>;
 
+// LANGMATCHES
+[[maybe_unused]] inline auto langMatching =
+    [](std::optional<std::string> optLanguageTag,
+       std::optional<std::string> optLanguageRange) {
+      if (!optLanguageTag.has_value() || !optLanguageRange.has_value()) {
+        return Id::makeUndefined();
+      }
+      // For LANGMATCHES(), the order of the arguments matter.
+      // The language-range "*" will match any non empty string
+      // (Sparql-standard).
+      const std::string& languageTag = optLanguageTag.value();
+      const std::string& languageRange = optLanguageRange.value();
+      if ((languageTag == "" && languageRange == "*") || languageRange == "") {
+        return Id::makeFromBool(false);
+      } else if (languageRange == "*") {
```

Comment:
```suggestion
      if ((languageTag.empty() || languageRange.empty() {
        return Id::makeFromBool(false);
      } else if (languageRange == "*") {
```
I think this is equivalent, but easier to read.